### ISSUE TITLE
can we add a user defined loss functions to our model?

### ISSUE THREAD

**Issue by susanth19** [patrikeh/go-deep#31]
Is there any way to use the user-defined loss functions so that the model can be trained based on the loss from these  defined functions? or am i missing something?

**Comment by patrikeh** [comment]
That isn't possible at the moment, although it would certainly make sense. In that case, loss functions would have to be passed as input to the trainer, with some sensible default based on the `Config.Mode`. Feel free to make a PR if you'd like.
